tests/test-telemetry-canary.py

```
@@ -439,6 +439,28 @@ def main():
         print("  ok    ingest: stub answering 403 exits non-zero")
 
+    # -------------------------------------------------------------
+    # 9b. Ingest: the diagnosis is status-specific, not "probably WAF" every time.
+    # A flat edge-block message sent a session hunting Cloudflare for a 400 that
+    # meant the deployed Worker was behind main. Story: #87.
+    # -------------------------------------------------------------
+    distinct = {}
+    for status, must_contain in (("400", "behind main"), ("401", "token"), ("403", "Cloudflare")):
+        _, stdout, stderr, _, _ = run_ingest_step(ingest_script, token=secret_token, curl_code=status)
+        combined = stdout + "\n" + stderr
+        error_line = next((l for l in combined.splitlines() if "::error::" in l), "")
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)